apps/rewards/app/components/Content/MyRewards.js

```diff
@@ -31,14 +31,70 @@ import {
 } from './RewardsTables'
 import { Empty } from '../Card'
 import { provideNetwork } from '../../../../../shared/ui'
+import { MILLISECONDS_IN_A_SECOND } from '../../../../../shared/ui/utils'
 
 const averageRewardsTitles = [ 'My Unclaimed Rewards', 'Year to Date', 'Inception to Date' ]
-// TODO: these need to be actually calculated
-const averageRewardsNumbers = [
-  formatAvgAmount(146, '+$', 'green'),
-  formatAvgAmount(19989.88, '$'),
-  formatAvgAmount(799.87, '$'),
-]
+
+const calculateMyRewardsSummary = (rewards, balances, convertRates) => {
+  return [
+    formatAvgAmount(calculateUnclaimedRewards(rewards, balances, convertRates),'+$', 'green'),
+    formatAvgAmount(calculateAllRewards(rewards, balances, convertRates), '$'),
+    formatAvgAmount(calculateYTDUserRewards(rewards, balances, convertRates), '$'),
+  ]
+}
+
+const calculateUnclaimedRewards = (rewards, balances, convertRates) => {
+  return sumUserRewards(
+    rewards,
+    balances,
+    convertRates,
+    (rew, bal) => !rew.claimed && rew.rewardToken === bal.address // rewardFilter
+  )
+}
+
+const calculateAllRewards = (rewards, balances, convertRates) => {
+  return sumUserRewards(
+    rewards,
+    balances,
+    convertRates,
+    (rew, bal) => rew.claimed && rew.rewardToken === bal.address, // RewardFilter
+  )
+}
+
+const calculateYTDUserRewards = (rewards, balances, convertRates) => {
+  const yearBeginning = new Date(new Date(Date.now()).getFullYear(), 0)
+  return sumUserRewards(
+    rewards,
+    balances,
+    convertRates,
+    (rew, bal) => rew.claimed && rew.rewardToken === bal.address && rew.endDate >= yearBeginning
+  )
+}
+
+const sumUserRewards = (rewards, balances, convertRates, rewardFilter) => {
+  return balances.reduce((balAcc, balance) => {
+    if (convertRates[balance.symbol]) {
+      return rewards.reduce((rewAcc,reward) => {
+        return (rewardFilter(reward, balance))
+          ?
+          rewAcc + reward.userRewardAmount / Math.pow(10, balance.decimals) / convertRates[balance.symbol]
```

Comment:
This might behave poorly with extreme values but assuming that nobody is getting a fiat payout that requires BigNumber management it's probably not a problem

Reply:
I've implemented BigNumber here to handle this better